openpype/settings/version_start.py

```diff
@@ -0,0 +1,39 @@
+import os
+
+from openpype.settings.lib import (
+    get_default_project_settings, get_project_settings
+)
+from openpype.lib.profiles_filtering import filter_profiles
+
+
+def get_versioning_start(
+    host="", task_name="", task_type="", family="", subset=""
```

Comment:
Base function should not use anything from current context and should define minimum required arguments + optional helper arguments. Then define another function which can use current context... There are more places where we must define the context instead of "guess it".

All functions that can use current context are in `openpype/pipeline/context_tools.py`.

```suggestion
    project_name,
    host_name,
    task_name,
    task_type,
    family,
    subset,
    project_settings=None
```

As I'm looking into the arguments, I don't know if it's good idea to have `family` and `subset` as arguments. It would not work for workfiles...

Reply:
For workfile we would input the family as `workfile` both for launching and publishing. Tested here and having only the workfile as v0 and all products (except workfile) become v1 when publishing works as well.
As for use case, I would try to be as flexible as possible cause studio might want to send workfiles as well as products to clients, so I dont want to limit this feature. I think the most common use case would be to just have `render` as v0, but again making this flexible means less code due to treating all versions the same.

Reply:
> Base function should not use anything from current context and should define minimum required arguments + optional helper arguments. Then define another function which can use current context for in-host usage... There are more places where we must define the context instead of "guess it".

Not 100% sure how to tackle this cause the settings are profiles, so minimum arguments we need will include context arguments for fetching a profile.
Otherwise base function would just return `1` if there is no context.

Reply:
> so minimum arguments we need will include context arguments for fetching a profile.

Exactly, but the function is getting project name from environment, which is wrong. You have to expect it as an argument.